File: Irrgarten-ruby/Irrgarten/game_state.rb
# frozen_string_literal: true
module Irrgarten
class GameState
  @labyrinthv
  @players
  @monsters
  @current_player
  @winner
  @log

  def initialize(l,p,m,cp,w,log_)
    @labyrinthv = l
    @players = p
    @monsters = m
    @current_player = cp
    @winner = w
    @log = log_
  end

  def labyrinthv
    return @labyrinthv
  end

  def players
    return @players
  end

  def monsters
    return @monsters
  end

  def current_players
    return @current_player
  end

  def winner
    return @winner
  end

  def log
    return @log
  end
end
end

File: Irrgarten-Java/src/irrgarten/GameState.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author dpant
 */
public class GameState {
    private String labyrinthv;
    private String players;
    private String monsters;
    private int currentPlayer;
    private boolean winner;
    private String log;
    
    GameState(String l, String p, String m, int cp, boolean w, String log_){
        labyrinthv = l;
        players = p;
        monsters = m;
        currentPlayer = cp;
        winner = w;
        log = log_;
    }
    
    public String getLabyrinthv(){ return labyrinthv;}
    public String getPlayers(){return players;}
    public String getMonsters(){return monsters;}
    public int getCurrentPlayer(){return currentPlayer;}
    public boolean getWinner(){return winner;}
    public String getLog(){return log;}
}
